server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestGetMappingAction.java

```diff
@@ -106,73 +107,95 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
-                                final List<String> matches = new ArrayList<>();
-                                outer:
-                                for (final String pattern : difference) {
-                                    if (pattern.contains("*")) {
-                                        for (final String typeName : typeNames) {
-                                            if (Regex.simpleMatch(pattern, typeName)) {
-                                                matches.add(pattern);
-                                                continue outer;
-                                            }
-                                        }
-                                    }
-                                }
-                                difference.removeAll(matches);
-
-                                final RestStatus status;
-                                builder.startObject();
-                                {
-                                    if (difference.isEmpty()) {
-                                        status = RestStatus.OK;
-                                    } else {
-                                        status = RestStatus.NOT_FOUND;
-                                        final String message = String.format(Locale.ROOT, "type" + (difference.size() == 1 ? "" : "s") +
-                                                " [%s] missing", Strings.collectionToCommaDelimitedString(difference));
-                                        builder.field("error", message);
-                                        builder.field("status", status.getStatus());
-                                    }
-                                    response.toXContent(builder, request);
-                                }
-                                builder.endObject();
-
-                                return new BytesRestResponse(status, builder);
-                            }
-                        }.onResponse(getMappingsResponse)));
+        final HttpChannel httpChannel = request.getHttpChannel();
+        return channel -> new RestCancellableNodeClient(client, httpChannel).admin().indices().getMappings(getMappingsRequest,
+                new DispatchingRestToXContentListener<>(threadPool.executor(ThreadPool.Names.MANAGEMENT), channel, request)
+                    .map(getMappingsResponse ->
+                        new RestGetMappingsResponse(types, getMappingsResponse, threadPool::relativeTimeInMillis, timeout)));
+    }
+
+    private static final class RestGetMappingsResponse implements StatusToXContentObject {
+        private final String[] types;
+        private final GetMappingsResponse response;
+        private final LongSupplier relativeTimeSupplierMillis;
+        private final TimeValue timeout;
+        private final long startTimeMs;
+        private final SortedSet<String> missingTypes;
+
+        private RestGetMappingsResponse(String[] types,
```

Comment:
nit: we call this `private` method from outside this inner class which means the compiler must generate a synthetic method to expose it. Best to drop the `private` instead.

Reply:
TIL 👍